devtools/datasette/publish.py

```diff
@@ -123,24 +118,27 @@ def deploy_datasette(deploy: str, fly_args: tuple[str]) -> int:
 
     python publish.py --fly -- --build-only
     """
-    pudl_out = PudlPaths().pudl_output
-    metadata_yml = metadata(pudl_out)
-    # Order the databases to highlight PUDL
-    datasets = (
+    pudl_output = PudlPaths().pudl_output
+    metadata_yml = create_datasette_metadata_yaml()
+
+    databases = (
         ["pudl.sqlite"]
-        + sorted(str(p.name) for p in pudl_out.glob("ferc*.sqlite"))
+        + sorted(str(p.name) for p in pudl_output.glob("ferc*.sqlite"))
         + ["censusdp1tract.sqlite"]
     )
-
+    # Make sure we have the expected metadata for databases
+    # headed to deployment.
+    if deploy != "metadata":
+        check_tables_have_metadata(metadata_yml, databases)
```

Comment:
I think this solves the entanglement issue you mentioned @zaneselvans . We only check if all of the databases tables have datasette metadata if we are actually deploy datatsette.

Reply:
The way the Datasette metadata is generated, and the checks rely on bespoke logic for constructing paths to the databases we expect to have metadata, and the dependence on the XBRL datapackage.json sometimes and the SQLite files other times and the PUDL Package other times all feels pretty fragile and janky, but that's beyond the scope of this PR.